shell/platform/android/io/flutter/embedding/engine/systemchannels/SpellCheckChannel.java

```diff
@@ -0,0 +1,120 @@
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import java.util.ArrayList;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+/**
+ * {@link SpellCheckChannel} is a platform channel that is used by the framework to initiate spell
+ * check in the embedding and for the embedding to send back the results.
+ *
+ * <p>When there is new text to be spell checked, the framework will send to the embedding the
+ * message {@code SpellCheck.initiateSpellCheck} with the {@code String} locale to spell check with
+ * and the {@code String} of text to spell check as arguments. In response, the {@link
+ * io.flutter.plugin.editing.SpellCheckPlugin} will make a call to Android's spell check service to
+ * fetch spell check results for the specified text.
+ *
+ * <p>Once the spell check results are received by the {@link
+ * io.flutter.plugin.editing.SpellCheckPlugin}, it will send to the framework the message {@code
+ * SpellCheck.updateSpellCheckResults} with the {@code ArrayList<String>} of encoded spell check
+ * results (see {@link
+ * io.flutter.plugin.editing.SpellCheckPlugin#onGetSentenceSuggestions(SentenceSuggestionsInfo[])}
+ * for details) with the text that these results correspond to appeneded to the front as an
+ * argument. For example, the argument may look like: {@code {"Hello, wrold!",
+ * "7.11.world\nword\nold"}}.
+ *
+ * <p>{@link io.flutter.plugin.editing.SpellCheckPlugin} implements {@link SpellCheckMethodHandler}
+ * to initiate spell check. Implement {@link SpellCheckMethodHandler} to respond to spell check
+ * requests.
+ */
+public class SpellCheckChannel {
+  private static final String TAG = "SpellCheckChannel";
+
+  public final MethodChannel channel;
+  private SpellCheckMethodHandler spellCheckMethodHandler;
+
+  private String textAwaitingResponse;
+  private MethodChannel.Result resultAwaitingResponse;
+
+  @NonNull
+  final MethodChannel.MethodCallHandler parsingMethodHandler =
+      new MethodChannel.MethodCallHandler() {
+        @Override
+        public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
+          if (spellCheckMethodHandler == null) {
+            Log.v(
+                TAG,
+                "No SpellCheckeMethodHandler registered, call not forwarded to spell check API.");
+            return;
+          }
+          String method = call.method;
+          Object args = call.arguments;
+          Log.v(TAG, "Received '" + method + "' message.");
+          switch (method) {
+            case "SpellCheck.initiateSpellCheck":
+              try {
+                final JSONArray argumentList = (JSONArray) args;
+                String locale = argumentList.getString(0);
+                String text = argumentList.getString(1);
+                textAwaitingResponse = text;
+                resultAwaitingResponse = result;
+
+                spellCheckMethodHandler.initiateSpellCheck(locale, text);
```

Comment:
what about just passing result?

```suggestion
                spellCheckMethodHandler.initiateSpellCheck(locale, text, result);
```